docs/intro.md

```diff
@@ -3,65 +3,176 @@
 FuTIL is an intermediate language and infrastructure for building compilers
 that generate custom hardware accelerators.
 
-The following instructions will install all dependencies required to build
-the compiler and run the compiler tests.
+FuTIL has many associated tools and depending on what you are doing, you
+probably only need to install a subset of them.
 
-Start the process by cloning the repository:
+We've organized this document into a core installation that is needed for everything
+and then add-on installations for our command-line driver tool, simulation, frontends,
+and synthesis.
+
+## Core Installation
+
+Start by cloning the repository:
 ```
 git clone https://github.com/cucapra/futil.git
 ```
 
 ### Compiler dependencies
 Install [Rust][rust] (it should automatically install `cargo`).
 
-### Testing dependencies
-The compiler uses expectation tests as well as hardware simulation tests.
+### Install tools for testing
+  1. [runt][] hosts our testing infrastructure. Install with:
+  `cargo install runt`
+  2. [jq][]:
+  Ubuntu: `sudo apt install jq`
+  Mac: `brew install jq`
+  Other platforms: [JQ installation][jq-install]
+
+### Verify installation
+Build the compiler with
+```
+cargo build
+```
+Then run the core tests with:
+```
+runt -i core
+```
+
+If everything has been installed correctly, this should not produce any failing
+tests.
+
+## Add-on Installations
+Here are instructions for optional add-ons. We recommend installing at least `fud`.
+### Fud (Command-line driver) [Recommended]
+<details><summary>Click to expand!</summary>
```

Comment:
Any reason to hide this? I assume all of the core stuff is covered before this and doesn't need any more abstracting.